Python rewrite

carcruz opened the pull request:
Epic: https://github.com/opentargets/issues/issues/3598

carcruz committed:
refactor: delete node app

carcruz committed:
feat: py ignore

carcruz committed:
feat: initial fastapi app

carcruz committed:
feat: parser and format

carcruz committed:
feat: formatter config

carcruz committed:
feat: intial fastapp

carcruz committed:
docs: update readme

carcruz committed:
feat: base app

carcruz committed:
feat: parser

carcruz committed:
feat: serviceworking

carcruz committed:
docker

carcruz committed:
update docker

carcruz committed:
feat: api env from file

carcruz committed:
feat: ensure docker running

carcruz committed:
feat: add health check endpoint and update OpenAI integration

carcruz committed:
refactor: simplify health check endpoint to return basic status

carcruz committed:
refactor: rename PublicationSummaryRequest to include payload structure and update logging to use new payload fields

carcruz committed:
fix: remove trailing slashes from publication endpoints for consistency

carcruz committed:
chore: update Dockerfile to use multi-stage builds and improve dependency management

carcruz committed:
feat: enhance FastAPI setup with custom documentation and static file serving

carcruz committed:
feat: enhance CORS configuration and input validation for publication models

carcruz committed:
Merge branch 'cc-py-rewrite' of github.com:opentargets/ot-ai-explain into cc-py-rewrite

remo87 commented on app/config.py:
```diff
@@ -0,0 +1,92 @@
+def get_config():
+    """Get configuration based on environment."""
+    env = os.getenv("APP_ENV", "development")
+    if env == "production":
```
These values should be configurable if we have a class ProductionConfig and DevelopmentConfig classes that would limit the environments that we can handle.

remo87 commented on app/main.py:
```diff
@@ -0,0 +1,207 @@
+    title="Open Targets AI API",
+    description=DESCRIPTION,
+    version="0.1.0",
+    debug=config.DEBUG,
```
This value should be configurable so that you can change the log level

remo87 commented on app/services/openai_service.py:
```diff
@@ -0,0 +1,112 @@
+# Configure module logger
+logger = logging.getLogger(__name__)
+
+logging.getLogger("openai").setLevel(logging.WARNING)
```
the logging level should be configurable

remo87 commented on app/services/openai_service.py:
```diff
@@ -0,0 +1,112 @@
+        structured_input = create_structured_input(target_symbol, disease_name, text)
+
+        response = client.responses.create(
+            model="gpt-5-mini",
```
should this value be configurable?

carcruz committed:
Refactor code structure for improved readability and maintainability

remo87 commented on app/services/openai_service.py:
```diff
@@ -0,0 +1,104 @@
+
+import logging
+
+from openai import OpenAI
```
we should use AsyncOpenAI to have an async call to the llm

remo87 commented on app/services/openai_service.py:
```diff
@@ -0,0 +1,104 @@
+config = get_config()
+
+# Initialize OpenAI client
+client = OpenAI(api_key=config.OPENAI_API_KEY)
```
we should change to AsyncOpenAI and a timeout

remo87 replied:
once we do this can handle multiple error code ```except BadRequestError as e:
    if "context_length" in str(e).lower() or "context length" in str(e).lower():
        raise HTTPException(status_code=413, detail="Publication too long for context")
    raise HTTPException(status_code=400, detail="LLM rejected the request")
except RateLimitError:
    raise HTTPException(status_code=503, detail="Upstream rate limit, retry later")
except (APITimeoutError, APIConnectionError):
    raise HTTPException(status_code=504, detail="LLM upstream timeout")```

remo87 commented on app/services/europe_pmc_service.py:
```diff
@@ -0,0 +1,498 @@
+# =============================================================================
+
+
+def fetch_publication_xml(pmc_id: str) -> str:
```
change this method to async

remo87 commented on app/services/europe_pmc_service.py:
```diff
@@ -0,0 +1,498 @@
+    try:
+        # Make HTTP request with timeout and appropriate headers
+        logger.info(f'Requesting publication XML for PMC ID: {pmc_id}')
+        response = requests.get(url, headers=REQUEST_HEADERS, timeout=REQUEST_TIMEOUT)
```
the http call should be async we could use httpx.AsyncClient

carcruz committed:
feat: update application name to 'Open Targets Platform AI API' and enhance description in main.py

remo87 commented on app/services/europe_pmc_service.py:
```diff
@@ -0,0 +1,498 @@
+        )
+
+
+def extract_publication_text(pmc_id: str, include_references: bool = False) -> str:
```
we should also change this to an async  `async def extract_publication_text`

remo87 commented on app/services/europe_pmc_service.py:
```diff
@@ -0,0 +1,498 @@
+    """
+    try:
+        # Step 1: Fetch XML content
+        xml_content = fetch_publication_xml(pmc_id)
```
we should await this call `await fetch_publication_xml(pmc_id)`

carcruz committed:
Refactor controllers and services to use async functions

- Updated `fetch_plain_text_from_europe_pmc` in `publication.py` to be asynchronous and adjusted service call accordingly.
- Modified `create_publication_summary` in `summary_controller.py` to be async and updated service call.
- Changed `get_publication_plain_text` in `main.py` to await the async function for fetching plain text.
- Refactored `handle_publication_summary_request` in `main.py` to be async and await the fetching of publication text.
- Updated `fetch_publication_xml` and `extract_publication_text` in `europe_pmc_service.py` to be async and use `httpx` for async HTTP requests.
- Changed `generate_publication_summary` in `openai_service.py` to be async and await the OpenAI API call.
- Removed `requests` dependency and added `httpx` to `pyproject.toml` and `uv.lock` for async HTTP handling.

carcruz committed:
feat: enhance OpenAI client initialization with timeout and retry settings

carcruz committed:
Add workflow_dispatch to build-and-push for GHCR-only manual pushes

Allows publishing an arbitrary tag to GHCR without touching GAR, useful
for pre-release images that don't warrant a GAR deploy yet.

remo87 committed:
update action version
